src/common/framework/resources.ts

```diff
@@ -5,11 +5,46 @@
  */
 let baseResourcePath = './resources';
 
+function getAbsoluteBaseResourcePath(path: string) {
+  const relparts = window.location.pathname.split('/');
+  relparts.pop();
+  const pathparts = path.split('/');
+
+  let i;
+  for (i = 0; i < pathparts.length; ++i) {
+    switch (pathparts[i]) {
+      case '':
+        break;
+      case '.':
+        break;
+      case '..':
+        relparts.pop();
+        break;
+      default:
+        relparts.push(pathparts[i]);
+        break;
+    }
+  }
+
+  return relparts.join('/');
+}
+
```

Comment:
Is it possible to just concatenate the strings rather than manually parsing out `.`/`..`?
For example, this URL works:
`https://www.google.com/some/path/../../index.html`
because the browser resolves `..` for us.

Reply:
Unfortunately, not possible. I tried with ts logic but it seems not work. But as you said, when you type `./..` in url bar, the page could be access correctly. I think this might be url bar handles the relative path.

Reply:
OK, makes sense, thanks.